app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/upscan/upscan.scala.html

```diff
@@ -0,0 +1,41 @@
+@*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import helper._
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.config.ViewConfig
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.UpscanService.UpscanServiceResponse._
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.routes
+
+@this(
+ mainTemplate: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.main_template,
+ pageHeading: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.components.page_heading
+)
+
+@(
+ descriptor: UpscanDescriptor
+)(implicit request: Request[_], messages: Messages, appConfig: ViewConfig)
+
+@mainTemplate(title = "Upload file", userType = None) {
+ @form(action = routes.UpscanController.upload(), 'enctype -> "multipart/form-data"){
+ @CSRF.formField
+  <input type="hidden" name="href" value="@descriptor.uploadRequest.href" />
```

Comment:
if we have the href here, someone could modify the HTML and upload something to whatever URL they put as the value here - is there a more secure way of getting the url, e.g. through mongo or something? Perhaps we include a UUID in the mongo document and submit that id instead of the url

Reply:
not sure what you mean here as well - I'm not sure how getting the URL out of mongo or elsewhere will make this more secure, you still have to render the form with that URL value. MITM attacks are mitigated by HTTPS/SSL.

Reply:
also this network traffic is within the platform VPC

Reply:
is this html page ever displayed to the user on the browser?

Reply:
ofc - it's the page which the user sees to choose the file and upload it

Reply:
it needs UX'ing by the UX team and then styled by Ali